src/lib/operators/abstract_operator.hpp

```diff
@@ -98,10 +106,14 @@ class AbstractOperator : public std::enable_shared_from_this<AbstractOperator>,
   // Calls set_transaction_context on itself and both input operators recursively
   void set_transaction_context_recursively(const std::weak_ptr<TransactionContext>& transaction_context);
 
-  // Returns a new instance of the same operator with the same configuration.
-  // Recursively copies the input operators.
-  // An operator needs to implement this method in order to be cacheable.
-  std::shared_ptr<AbstractOperator> deep_copy() const;
+  /**
+   * Returns a new instance of the same operator with the same configuration.
+   * Recursively copies the input operators. An operator needs to implement this function in order to be cacheable.
+   * Subplans are deduplicated automatically using @param copied_ops.
```

Comment:
Is this really "deduplication"? We do not check for duplicates in the input PQP, we just make sure that the output PQP does not introduce additional duplicates.

Also, this would be another place where I would like to reference that canonical extended comment about copied_ops.

Reply:
Good point. We can only preserve the deduplication of input PQPs.